<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Общая-информация-о-данных" data-toc-modified-id="Общая-информация-о-данных-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Общая информация о данных</a></span></li><li><span><a href="#Предобработка-данных" data-toc-modified-id="Предобработка-данных-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Предобработка данных</a></span><ul class="toc-item"><li><span><a href="#Определение-и-заполнение-пропущенных-значений" data-toc-modified-id="Определение-и-заполнение-пропущенных-значений-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Определение и заполнение пропущенных значений</a></span></li><li><span><a href="#Замена-вещественного-типа-данных-на-целочисленный" data-toc-modified-id="Замена-вещественного-типа-данных-на-целочисленный-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>Замена вещественного типа данных на целочисленный</a></span></li><li><span><a href="#Удаление-дубликатов" data-toc-modified-id="Удаление-дубликатов-2.3"><span class="toc-item-num">2.3&nbsp;&nbsp;</span>Удаление дубликатов</a></span></li><li><span><a href="#Выделение-лемм-в-значениях-столбца-purpose---&quot;цели-кредита&quot;" data-toc-modified-id="Выделение-лемм-в-значениях-столбца-purpose---&quot;цели-кредита&quot;-2.4"><span class="toc-item-num">2.4&nbsp;&nbsp;</span>Выделение лемм в значениях столбца <code>purpose</code> - "цели кредита"</a></span></li><li><span><a href="#Категории-данных" data-toc-modified-id="Категории-данных-2.5"><span class="toc-item-num">2.5&nbsp;&nbsp;</span>Категории данных</a></span></li></ul></li><li><span><a href="#Гипотезы" data-toc-modified-id="Гипотезы-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Гипотезы</a></span><ul class="toc-item"><li><span><a href="#Гипотеза-о-зависимости-между-наличием-детей-и-возвратом-кредита-в-срок" data-toc-modified-id="Гипотеза-о-зависимости-между-наличием-детей-и-возвратом-кредита-в-срок-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>Гипотеза о зависимости между наличием детей и возвратом кредита в срок</a></span></li><li><span><a href="#Гипотеза-о-зависимости-между-семейным-положением-и-возвратом-кредита-в-срок" data-toc-modified-id="Гипотеза-о-зависимости-между-семейным-положением-и-возвратом-кредита-в-срок-3.2"><span class="toc-item-num">3.2&nbsp;&nbsp;</span>Гипотеза о зависимости между семейным положением и возвратом кредита в срок</a></span></li><li><span><a href="#Гипотеза-о-зависимости-между-уровнем-дохода-и-возвратом-кредита-в-срок" data-toc-modified-id="Гипотеза-о-зависимости-между-уровнем-дохода-и-возвратом-кредита-в-срок-3.3"><span class="toc-item-num">3.3&nbsp;&nbsp;</span>Гипотеза о зависимости между уровнем дохода и возвратом кредита в срок</a></span></li><li><span><a href="#Гипотеза-о-влиянии-цели-кредита-на-его-возврат-в-срок" data-toc-modified-id="Гипотеза-о-влиянии-цели-кредита-на-его-возврат-в-срок-3.4"><span class="toc-item-num">3.4&nbsp;&nbsp;</span>Гипотеза о влиянии цели кредита на его возврат в срок</a></span></li></ul></li><li><span><a href="#Общий-вывод" data-toc-modified-id="Общий-вывод-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Общий вывод</a></span></li></ul></div>

## Общая информация о данных

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('/datasets/data.csv')

In [3]:
df.head(15)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   

     total_income                            purpose  
0   253875.639453                      покупка жилья  
1   112080.014102            приобретение автомобиля  
2   145885.952297                      покупка жилья  
3   267628.550329         дополнительное образование  
4   158616.077870                    сыграть свадьбу  
5   255763.565419                      покупка жилья  
6   240525.971920                  операции с жильем  
7   135823.934197                        образование  
8    95856.832424              на проведение свадьбы  
9   144425.938277            покупка жилья для семьи  
10  113943.491460               покупка недвижимости  
11   77069.234271  покупка коммерческой недвижимости  
12            NaN                    сыграть свадьбу  
13  130458.228857            приобретение автомобиля  
14  165127.911772         покупка жилой недвижимости

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
children            21525 non-null int64
days_employed       19351 non-null float64
dob_years           21525 non-null int64
education           21525 non-null object
education_id        21525 non-null int64
family_status       21525 non-null object
family_status_id    21525 non-null int64
gender              21525 non-null object
income_type         21525 non-null object
debt                21525 non-null int64
total_income        19351 non-null float64
purpose             21525 non-null object
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


**Вывод** - в столбцах `days_employed`и `total_income` есть пропущенные значения

## Предобработка данных

### Определение и заполнение пропущенных значений

In [5]:
df.isna().sum()





children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

In [6]:
df = df.fillna(0)

df[df['days_employed'] == 0]


children  days_employed  dob_years education  education_id  \
12            0            0.0         65   среднее             1   
26            0            0.0         41   среднее             1   
29            0            0.0         63   среднее             1   
41            0            0.0         50   среднее             1   
55            0            0.0         54   среднее             1   
...         ...            ...        ...       ...           ...   
21489         2            0.0         47   Среднее             1   
21495         1            0.0         50   среднее             1   
21497         0            0.0         48    ВЫСШЕЕ             0   
21502         1            0.0         42   среднее             1   
21510         2            0.0         28   среднее             1   

               family_status  family_status_id gender  income_type  debt  \
12          гражданский брак                 1      M    пенсионер     0   
26           женат / замужем                 0      M  госслужащий     0   
29     Не женат / не замужем                 4      F    пенсионер     0   
41           женат / замужем                 0      F  госслужащий     0   
55          гражданский брак                 1      F    пенсионер     1   
...                      ...               ...    ...          ...   ...   
21489        женат / замужем                 0      M    компаньон     0   
21495       гражданский брак                 1      F    сотрудник     0   
21497        женат / замужем                 0      F    компаньон     0   
21502        женат / замужем                 0      F    сотрудник     0   
21510        женат / замужем                 0      F    сотрудник     0   

       total_income                           purpose  
12              0.0                   сыграть свадьбу  
26              0.0                       образование  
29              0.0  строительство жилой недвижимости  
41              0.0  сделка с подержанным автомобилем  
55              0.0                   сыграть свадьбу  
...             ...                               ...  
21489           0.0              сделка с автомобилем  
21495           0.0                           свадьба  
21497           0.0        строительство недвижимости  
21502           0.0  строительство жилой недвижимости  
21510           0.0           приобретение автомобиля  

[2174 rows x 12 columns]

In [7]:
df[(df['days_employed'] != 0) & (df['total_income'] == 0)]

Empty DataFrame
Columns: [children, days_employed, dob_years, education, education_id, family_status, family_status_id, gender, income_type, debt, total_income, purpose]
Index: []

In [8]:
df[df['days_employed'] == 0]['income_type'].value_counts()


сотрудник          1105
компаньон           508
пенсионер           413
госслужащий         147
предприниматель       1
Name: income_type, dtype: int64

In [9]:
df['income_type'].value_counts()

сотрудник          11119
компаньон           5085
пенсионер           3856
госслужащий         1459
предприниматель        2
безработный            2
в декрете              1
студент                1
Name: income_type, dtype: int64

**Вывод:** Объём данных с пропущенными значениями состовляет 10% от общего количества и просто удалить строки будет неразумно.
Т.к. только для гипотезы о связи между доходом и возвратом кредита нам нужны данные о `total_income` — "ежемесячный доход", а данные `days_employed` — "общий трудовой стаж в днях" в гипотезах не принимает участие, то заменим все пропуски на ноль. 
А для `total_income` будем учитывать, что ноль - это отсутсвие данных. **Причины пропусков:** от действий операторов, например, когда данные в соотвествующей ячейке были выделены, нажали на цифровой клавиатуре клавишу 0 случайно, до технических ошибок при перезаписи данных.

### Замена вещественного типа данных на целочисленный

In [10]:
df['days_employed'] = df['days_employed'].astype('int')

df['total_income'] = df['total_income'].astype('int')

df.info()

#df.head(15)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
children            21525 non-null int64
days_employed       21525 non-null int64
dob_years           21525 non-null int64
education           21525 non-null object
education_id        21525 non-null int64
family_status       21525 non-null object
family_status_id    21525 non-null int64
gender              21525 non-null object
income_type         21525 non-null object
debt                21525 non-null int64
total_income        21525 non-null int64
purpose             21525 non-null object
dtypes: int64(7), object(5)
memory usage: 2.0+ MB


### Удаление дубликатов

In [11]:
df.duplicated().sum()

54

In [12]:
df = df.drop_duplicates().reset_index(drop = True)

In [13]:
df.duplicated().sum()

0

Явные дубликаты ушли в прошлое.

In [14]:
#применим метод str.lower() ко всем столбцам имеющим тип object

name_columns = df.columns

for name_column in name_columns:
    if df[name_column].dtypes == 'object':
        df[name_column] = df[name_column].str.lower()

In [15]:
df.duplicated().sum()

17

Привели к единообразному виду и ещё 17 дубликатов нашли.

In [16]:
df = df.drop_duplicates().reset_index(drop = True)

In [17]:
df.duplicated().sum()

0

In [18]:
df.head(10)

children  days_employed  dob_years education  education_id  \
0         1          -8437         42    высшее             0   
1         1          -4024         36   среднее             1   
2         0          -5623         33   среднее             1   
3         3          -4124         32   среднее             1   
4         0         340266         53   среднее             1   
5         0           -926         27    высшее             0   
6         0          -2879         43    высшее             0   
7         0           -152         50   среднее             1   
8         2          -6929         35    высшее             0   
9         0          -2188         41   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      f   сотрудник     0        253875   
1   женат / замужем                 0      f   сотрудник     0        112080   
2   женат / замужем                 0      m   сотрудник     0        145885   
3   женат / замужем                 0      m   сотрудник     0        267628   
4  гражданский брак                 1      f   пенсионер     0        158616   
5  гражданский брак                 1      m   компаньон     0        255763   
6   женат / замужем                 0      f   компаньон     0        240525   
7   женат / замужем                 0      m   сотрудник     0        135823   
8  гражданский брак                 1      f   сотрудник     0         95856   
9   женат / замужем                 0      m   сотрудник     0        144425   

                      purpose  
0               покупка жилья  
1     приобретение автомобиля  
2               покупка жилья  
3  дополнительное образование  
4             сыграть свадьбу  
5               покупка жилья  
6           операции с жильем  
7                 образование  
8       на проведение свадьбы  
9     покупка жилья для семьи

### Выделение лемм в значениях столбца `purpose` - "цели кредита"

In [19]:
df_unique = df['purpose'].value_counts()

df_unique
#len(df_unique)

свадьба                                   791
на проведение свадьбы                     768
сыграть свадьбу                           765
операции с недвижимостью                  675
покупка коммерческой недвижимости         661
операции с жильем                         652
покупка жилья для сдачи                   651
операции с коммерческой недвижимостью     650
покупка жилья                             646
жилье                                     646
покупка жилья для семьи                   638
строительство собственной недвижимости    635
недвижимость                              633
операции со своей недвижимостью           627
строительство жилой недвижимости          624
покупка недвижимости                      621
покупка своего жилья                      620
строительство недвижимости                619
ремонт жилью                              607
покупка жилой недвижимости                606
на покупку своего автомобиля              505
заняться высшим образованием      

In [20]:
from pymystem3 import Mystem
m = Mystem()

purpose_lemmas = []

for i in range(50):
    lemmas = m.lemmatize(df.loc[i,'purpose'])
    purpose_lemmas.append(lemmas)
    

In [21]:
from collections import Counter

Counter(purpose_lemmas[49])

Counter({'автомобиль': 1, '\n': 1})

### Категории данных

In [23]:
df.head(10)

children  days_employed  dob_years education  education_id  \
0         1          -8437         42    высшее             0   
1         1          -4024         36   среднее             1   
2         0          -5623         33   среднее             1   
3         3          -4124         32   среднее             1   
4         0         340266         53   среднее             1   
5         0           -926         27    высшее             0   
6         0          -2879         43    высшее             0   
7         0           -152         50   среднее             1   
8         2          -6929         35    высшее             0   
9         0          -2188         41   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      f   сотрудник     0        253875   
1   женат / замужем                 0      f   сотрудник     0        112080   
2   женат / замужем                 0      m   сотрудник     0        145885   
3   женат / замужем                 0      m   сотрудник     0        267628   
4  гражданский брак                 1      f   пенсионер     0        158616   
5  гражданский брак                 1      m   компаньон     0        255763   
6   женат / замужем                 0      f   компаньон     0        240525   
7   женат / замужем                 0      m   сотрудник     0        135823   
8  гражданский брак                 1      f   сотрудник     0         95856   
9   женат / замужем                 0      m   сотрудник     0        144425   

                      purpose  
0               покупка жилья  
1     приобретение автомобиля  
2               покупка жилья  
3  дополнительное образование  
4             сыграть свадьбу  
5               покупка жилья  
6           операции с жильем  
7                 образование  
8       на проведение свадьбы  
9     покупка жилья для семьи

**Создадим** два словаря `education_dict` и `family_status_dict`, т.к. есть и обозначение и идентификационный номер.

In [24]:
education_dict = df[['education','education_id']]
#education_dict.head(15)

In [25]:
education_dict = education_dict.drop_duplicates().reset_index(drop=True)

education_dict.head(15)

education  education_id
0               высшее             0
1              среднее             1
2  неоконченное высшее             2
3            начальное             3
4       ученая степень             4

In [26]:
df['education'].value_counts() # проверка набора данных в словаре

среднее                15172
высшее                  5250
неоконченное высшее      744
начальное                282
ученая степень             6
Name: education, dtype: int64

In [27]:
family_status_dict = df[['family_status','family_status_id']]

In [28]:
family_status_dict = family_status_dict.drop_duplicates().reset_index(drop=True)

family_status_dict.head(15)

family_status  family_status_id
0        женат / замужем                 0
1       гражданский брак                 1
2         вдовец / вдова                 2
3              в разводе                 3
4  не женат / не замужем                 4

In [29]:
df['family_status'].value_counts() # проверка набора данных в словаре

женат / замужем          12339
гражданский брак          4151
не женат / не замужем     2810
в разводе                 1195
вдовец / вдова             959
Name: family_status, dtype: int64

In [30]:
len(df['total_income'].value_counts())

18607

In [31]:
df['purpose'].value_counts()

свадьба                                   791
на проведение свадьбы                     768
сыграть свадьбу                           765
операции с недвижимостью                  675
покупка коммерческой недвижимости         661
операции с жильем                         652
покупка жилья для сдачи                   651
операции с коммерческой недвижимостью     650
покупка жилья                             646
жилье                                     646
покупка жилья для семьи                   638
строительство собственной недвижимости    635
недвижимость                              633
операции со своей недвижимостью           627
строительство жилой недвижимости          624
покупка недвижимости                      621
покупка своего жилья                      620
строительство недвижимости                619
ремонт жилью                              607
покупка жилой недвижимости                606
на покупку своего автомобиля              505
заняться высшим образованием      

**Для проверки гипотез целесообразно будет:**
1. Провести классификацию по `total_income` — ежемесячный доход с созданием словаря, т.к. уникальных записей в данном столбце 18 607.
2. Провести классификацию по `purpose` — цель получения кредита с созданием словаря, т.к. есть цели имеющие одинаковый смысл, но выраженные разнами словами или их сочетаниями.



**Категоризация по детям**

In [32]:
df_children_debt = df[['children','debt']].groupby('children').agg({'debt': ['sum','count']})

df_children_debt['sum/count_%'] = (df_children_debt['debt']['sum'] / df_children_debt['debt']['count']) * 100

df_children_debt

debt        sum/count_%
           sum  count            
children                         
-1           1     47    2.127660
 0        1063  14091    7.543822
 1         444   4808    9.234609
 2         194   2052    9.454191
 3          27    330    8.181818
 4           4     41    9.756098
 5           0      9    0.000000
 20          8     76   10.526316

Что мы видим? В столбце children указано количество детей, а не просто идентификатор есть-дети/нет-детей. По моему мнению, значение "-1" это ошибка переноса данных из одной системы в другую и это означает один ребенок. 20 может быть ошибкой от преобразования 2.0 в 2 или может быть числом 20, есть же приёмные семьи. В любом случае приведём к двум категориям "есть дети" или "нет детей". Для осуществления задуманного напишем функцию аvailability_children которая будет возращать нет детей если значение в столбце children равно 0 и есть дети в любом другом случае. Далее методом apply() создадим таблицу из трёх колонок children, debt и аv_children.

In [33]:
def аvailability_children(children):
    if children == 0:
        return 'нет детей'
    return 'есть дети'

In [34]:
df_children_debt = df[['children','debt']]

df_children_debt = df_children_debt.copy()

df_children_debt['аv_children'] = df_children_debt['children'].apply(аvailability_children)

df_children_debt.head()

children  debt аv_children
0         1     0   есть дети
1         1     0   есть дети
2         0     0   нет детей
3         3     0   есть дети
4         0     0   нет детей

**Котегоризация по уровню дохода**

In [35]:
df_total_income_debt = df[['total_income','debt']]

Итак, придумываем группы.

1. Со значением ноль будет назаваться - нет данных
2. Со значениями в диапазоне от 20 000 до 60 000 будет нываться - на жизнь хватает
3. Со значениями в диапазоне от 60 001 до 145 017 будет нываться - неплохо живём
4. Со значениями в диапазоне от 145 018 до 350 000 будет нываться - хорошо живём
5. Со значениями в диапазоне от 350 001 до 1 000 000 будет нываться - ЖивЁм
6. Со значениями в диапазоне от 1 000 001 до 2 265 700 будет нываться - трудоголики

In [36]:
def total_income_group(total_income):
    """
    Возвращает группу по значению ежемесячного дохода total_income, используя правила описанные ячейкой выше.
    
    """
    if total_income == 0:
        return 'нет данных'
    if total_income <= 60000:
        return 'на жизнь хватает'
    if total_income <= 145017:
        return 'неплохо живём'
    if total_income <= 350000:
        return 'хорошо живём'
    if total_income <= 1000000:
        return 'ЖивЁм'
    return 'трудоголики'

In [37]:
#тест функции
total_income_group(1028542)

'трудоголики'

In [38]:
df_total_income_debt = df_total_income_debt.copy()

df_total_income_debt['total_income_group'] = df_total_income_debt['total_income'].apply(total_income_group)

df_total_income_debt.head()

total_income  debt total_income_group
0        253875     0       хорошо живём
1        112080     0      неплохо живём
2        145885     0       хорошо живём
3        267628     0       хорошо живём
4        158616     0       хорошо живём

**Группировка по целям кредитования**

In [39]:
df_purpose_debt = df[['purpose','debt']]
df_purpose_debt['purpose'].value_counts()

свадьба                                   791
на проведение свадьбы                     768
сыграть свадьбу                           765
операции с недвижимостью                  675
покупка коммерческой недвижимости         661
операции с жильем                         652
покупка жилья для сдачи                   651
операции с коммерческой недвижимостью     650
покупка жилья                             646
жилье                                     646
покупка жилья для семьи                   638
строительство собственной недвижимости    635
недвижимость                              633
операции со своей недвижимостью           627
строительство жилой недвижимости          624
покупка недвижимости                      621
покупка своего жилья                      620
строительство недвижимости                619
ремонт жилью                              607
покупка жилой недвижимости                606
на покупку своего автомобиля              505
заняться высшим образованием      

Выделим пять групп и напишем функцию для проведения группировки.

1. группа - свадьба
2. группа - жильё
3. группа - недвижимость
4. группа - автомобиль
5. группа - образование

In [40]:
def purpose_group(purpose):
    """
    Возвращает группу по значению цели кредита.
    
    """
    if 'свадьб' in purpose:
        return 'свадьба'
    if 'жил' in purpose:
        return 'жильё'
    if 'недвижимост' in purpose:
        return 'недвижимость'
    if 'автомобил' in purpose:
        return 'автомобиль'
    return 'образование'

In [41]:
#тест функции
purpose_group('хочу милый сердцу автомобиль')

'автомобиль'

In [42]:
df_purpose_debt = df_purpose_debt.copy()
df_purpose_debt['purpose_group'] = df_purpose_debt['purpose'].apply(purpose_group)
df_purpose_debt.head(10)

purpose  debt purpose_group
0               покупка жилья     0         жильё
1     приобретение автомобиля     0    автомобиль
2               покупка жилья     0         жильё
3  дополнительное образование     0   образование
4             сыграть свадьбу     0       свадьба
5               покупка жилья     0         жильё
6           операции с жильем     0         жильё
7                 образование     0   образование
8       на проведение свадьбы     0       свадьба
9     покупка жилья для семьи     0         жильё

## Гипотезы

### Гипотеза о зависимости между наличием детей и возвратом кредита в срок

In [46]:
df_children_debt_group = df_children_debt.groupby('аv_children').agg({'debt': ['sum','count']})

df_children_debt_group['sum/count_%'] = (df_children_debt_group['debt']['sum'] / df_children_debt_group['debt']['count']) * 100

df_children_debt_group

debt        sum/count_%
              sum  count            
аv_children                         
есть дети     678   7363    9.208203
нет детей    1063  14091    7.543822

В абсолютном выражении людей со сложными жизненными обстоятельствами больше в группе без детей чем в группе с детьми, но их и как заёмщиков больше в два раза. Если же сравнивать доли "сложных людей" в той и другой группе, то мы видим, что разницы нет  практически. Отсюда делаем вывод, что зависимости между наличием детей и возвратом кредита в срок - не наблюдается. 

### Гипотеза о зависимости между семейным положением и возвратом кредита в срок

In [47]:
df_family_status_debt = df[['family_status','debt']].groupby('family_status').agg({'debt': ['sum','count']})

df_family_status_debt['sum/count_%'] = (df_family_status_debt['debt']['sum'] / df_family_status_debt['debt']['count']) * 100

df_family_status_debt.sort_values(by = 'sum/count_%',ascending = False)


debt        sum/count_%
                       sum  count            
family_status                                
не женат / не замужем  274   2810    9.750890
гражданский брак       388   4151    9.347145
женат / замужем        931  12339    7.545182
в разводе               85   1195    7.112971
вдовец / вдова          63    959    6.569343

По имеющимся данным к сожелению не получится выдвинуть теорию что в состоянии выбора предпочтения стоит отдавать семейному заёмщику, слишкам мала дифференциация в столбце `sum/count_%`. Самые не дисциплинированные оказаль холостые представители рода человеческого, что возможно объясняется тем фактом, что отсутствие постоянного пратнёра вызвано отсутсвием желания брать на себя дополнительную ответсвенность перед кем-то другим, что соответсвенно распростроняется и на кредитора.

### Гипотеза о зависимости между уровнем дохода и возвратом кредита в срок

In [55]:
df_total_income_debt = df_total_income_debt.groupby('total_income_group').agg({'debt': ['sum','count']})

df_total_income_debt['sum/count_%'] = (df_total_income_debt['debt']['sum'] / df_total_income_debt['debt']['count']) * 100

df_total_income_debt.sort_values(by = 'sum/count_%',ascending = False)


debt       sum/count_%
                    sum count            
total_income_group                       
неплохо живём       755  8870    8.511838
нет данных          170  2103    8.083690
хорошо живём        712  8817    8.075309
трудоголики           2    25    8.000000
ЖивЁм                53   833    6.362545
на жизнь хватает     49   806    6.079404

И что мы видим? Значимых различий нет. Распределение вполне объяснимое, после того как его видишь.

### Гипотеза о влиянии цели кредита на его возврат в срок

In [60]:
df_purpose_debt = df_purpose_debt.groupby('purpose_group').agg({'debt': ['sum','count']})

df_purpose_debt['sum/count_%'] = (df_purpose_debt['debt']['sum'] / df_purpose_debt['debt']['count']) * 100

df_purpose_debt.sort_values(by = 'sum/count_%',ascending = False)

debt       sum/count_%
               sum count            
purpose_group                       
автомобиль     403  4306    9.359034
образование    370  4013    9.220035
свадьба        186  2324    8.003442
недвижимость   385  5121    7.518063
жильё          397  5690    6.977153

И здесь, без сюрпризов. Жильё - это безопасность, основательность тут не до хулиганства с просрочкой возврата части кредита. Автомобиль - это свобода, что возможно несколько расслабляет. 

## Общий вывод

1. Причины по которым заёмщики позволяют себе допускать несвоевременное погащение части кредита, в представленных нам данных, не обнаружены, её надо искать в другом месте. Скажем так, проведя актетирование среди тех кто допускает несоблюдение графика плотежей, и выяснив причины такого поведения. Если эти причины поддаются формальному определению например, `работа вахтовым методом`, (а не что-то в духе "мне сказал мой внутренний голос - сегодня не плати, день неподходящий", т.е. ввести причину `внутриличностные мотивы` не дадут, начальство не поймёт, к сожелению), то первые три вводим как обязательные при заполнении данных о заёмщике, в случае если их там нет ещё.

2. В каждой подгруппе риск неплатеже не преващает 10%. Приемлемо это или нет предстоит решать лицам принимающим решения. Успехов им удачи и терпения на этом пути. Если в последнем пожелании и есть сарказм, то честно, честно совсев чуть-чуть.
